src/Clover/Provider.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace SocialiteProviders\Clover;
+
+use GuzzleHttp\RequestOptions;
+use Illuminate\Support\Arr;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    const IDENTIFIER = 'CLOVER';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = [''];
+
+    /**
+     * Indicates if the session state should be utilized.
+     *
+     * @var bool
+     */
+    protected $stateless = true;
+
+    protected function getApiDomain(): string
+    {
+        return match (true) {
+            config('services.clover.sandbox-mode') => 'sandbox.dev.clover.com',
+            default => 'www.clover.com',
+        };
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase('https://'.$this->getApiDomain().'/oauth/v2/authorize', $state);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenUrl()
+    {
+        $domain = match (true) {
+            config('services.clover.sandbox-mode') => 'apisandbox.dev.clover.com',
+            default => 'api.clover.com',
+        };
+
+        return 'https://'.$domain.'/oauth/token?'.Arr::query([
+            'client_id' => config('services.clover.client_id'),
+            'client_secret' => config('services.clover.client_secret'),
+            'code' => $this->getCode(),
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getUserByToken($token)
+    {
+        $requestParams = str(request()->fullUrl())
+            ->after('?')
+            ->explode('&')
+            ->mapWithKeys(fn (string $keyAndValue) => [str($keyAndValue)->before('=')->toString() => str($keyAndValue)->after('=')->toString()]);
```

Comment:
Huh, this is strange. The data comes back in the URL?

Reply:
yeah…it’s kinda weird